thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AnomaliesResource.java

```diff
@@ -784,29 +783,76 @@ private AnomalyDetails constructAnomalyDetails(String metricName, String dataset
     return anomalyDetails;
   }
 
+  // this function generate the time ragne for viewing purpose.
+  private TimeRange getTimeseriesViewingOffsetedTimes(long anomalyStartTime, long anomalyEndTime, DatasetConfigDTO datasetConfig) {
+    TimeUnit dataTimeunit = datasetConfig.getTimeUnit();
+    Period preOffsetPeriod; // The offset we view before the anomaly
+    Period postOffsetPeriod; // The offset we view after the anomaly
+    switch (dataTimeunit) {
+      case DAYS: // 30 days. User needs to see the trend 30 days ahead
```

Comment:
We're creating too many custom Period objects, and we could mistake positions when reading/editing. There's a method in TimeGranularity called toPeriod. Try to see if you can use that, or create a static method in utils Period toPeriod(int timeSize, TimeUnit timeUnit).